draft-ietf-jsonpath-base.md

```diff
@@ -156,77 +156,81 @@ their UTF-8 encoding.
 For example, the Unicode PLACE OF INTEREST SIGN (U+2318) would be defined
 in ABNF as `%x2318`.
 
+The terminology of {{-json}} applies except where clarified below.
+Definitions for "Object", "Array", and "Number" remain unchanged.
+Importantly "object" and "array" in particular do not take on a
+generic meaning, such as they would in a general programming context.
 
-The terminology of {{-json}} applies.
+Additional terms used in this specification are defined below.
 
-For the purposes of this specification, a JSON value as defined by
-{{-json}} is also viewed as a tree of nodes.
-Each node holds a JSON value of one of the types defined in {{-json}};
-further nodes within the JSON value are the elements of arrays and the
-member values of JSON objects contained in the JSON value and are
-themselves JSON values of one of the types defined in {{-json}}.
-(The type of the JSON value held by a node is
-sometimes referred to as the type of the node.)
-
-A JSONPath query is applied to a JSON value that is supplied to it as
-its Argument.
-The node referring to the entirety of this JSON value is also
-referred to as its root node.
-
-JSON value:
+Value:
 : As per {{-json}}, a structure complying to the generic data model of JSON, i.e.,
   composed of components such as containers, namely JSON objects and arrays, and
   atomic data, namely null, true, false, numbers, and text strings.
 
-Node:
-: A JSON value, with an emphasis on its location within the
-  Argument.  I.e., a JSON value that is identical to or contained within the JSON
-  value to which the query is applied.  A node can be viewed as a
-  combination of a (1) JSON value and (2) its location in the
-  Argument; the latter can, if desired, be represented as an Output Path.
-
-Object:
-: A JSON object as defined in {{-json}}.
-  Never used in its generic sense, e.g., for programming language objects.
+String:
```

Comment:
That is *NOT* what a string is for the purpose of this specification.
We do not need to confuse our readers.
(It also is completely unrelated to the JSON syntax.)

Reply:
Maybe you should try offering an alternative because this is PRECISELY what a string is for JSON Path: a single- or double-quoted sequence of characters.  JSON uses UTF-8, so that's what I proposed.  If you think it can be better, how so?